frontend/src/pages/modelRegistry/screens/ModelVersionDetails/ModelVersionDetails.tsx

```diff
@@ -41,6 +47,26 @@ const ModelVersionsDetails: React.FC<ModelVersionsDetailProps> = ({ tab, ...page
     servingRuntimes.refresh();
   }, [inferenceServices, servingRuntimes, refreshModelVersion]);
 
+  useEffect(() => {
+    if (rm?.state === ModelState.ARCHIVED && mv?.id) {
+      navigate(
+        archiveModelVersionDetailsUrl(
+          mv.id,
+          mv.registeredModelId,
+          preferredModelRegistry?.metadata.name,
+        ),
+      );
+    } else if (mv?.state === ModelState.ARCHIVED) {
+      navigate(
+        modelVersionArchiveDetailsUrl(
+          mv.id,
+          mv.registeredModelId,
+          preferredModelRegistry?.metadata.name,
+        ),
+      );
+    }
+  });
```

Comment:
Should all these new useEffects have dependency arrays? I'm not sure whether it matters since the `if` conditions are the same perf impact as comparing dependencies across renders, but it may still be better to not run the effects every render in case we modify them later.